trayButton/tray-button.c

```diff
@@ -832,21 +832,33 @@ int CALLBACK WinMain(HINSTANCE hInst, HINSTANCE hPrevInst, LPSTR cmd, int show)
 		currentDpi = getDpi(taskbar);
 		checkHighContrast();
 
-		// Create the button window
-		buttonWindow = CreateWindowEx(
-			WS_EX_TRANSPARENT,
-			BUTTON_CLASS,
-			BUTTON_CLASS,
-			WS_VISIBLE | WS_CHILD | WS_CLIPSIBLINGS | WS_TABSTOP,
-			0, 0, BUTTON_WIDTH, 40,
-			taskbar,
-			null,
-			0,
-			null);
-
-		if (!buttonWindow) {
-			fail("CreateWindowEx");
-		}
+		DWORD lastError = 0;
+		do {
+			// Create the button window
+			buttonWindow = CreateWindowEx(
+				WS_EX_TRANSPARENT,
+				BUTTON_CLASS,
+				BUTTON_CLASS,
+				WS_VISIBLE | WS_CHILD | WS_CLIPSIBLINGS | WS_TABSTOP,
+				0, 0, BUTTON_WIDTH, 40,
+				taskbar,
+				null,
+				0,
+				null);
+
+			if (!buttonWindow) {
+				// Sometimes CreateWindowEx fails with ERROR_ACCESS_DENIED (can be reproduced if the start menu is open)
+				// Also, creating without a parent but using SetParent returns ERROR_INVALID_PARAMETER, under the same
+				// conditions.
+				DWORD thisError = GetLastError();
+				if (thisError != lastError) {
+					fail("CreateWindowEx (retrying)");
+					lastError = thisError;
+				}
+				Sleep(1000);
+			}
+			// Continue trying to create the window if it didn't succeed.
```

Comment:
Maybe stupid question, but, what if we never succeed in creating the tray button? 
Bear in mind that my comment goes beyond saying "let's add a timeout" and raise an error if we can't create the tray icon after X retries. Is it acceptable for gpii-app to keep starting up even if we can't create the tray icon? 
Of course, this is just an observation, not blocking the merge of this PR at all, but maybe something that we should think about it in case we find out that even with this patch, we can't guarantee 100% that the tray icon is going to be there. :man_shrugging:

Reply:
I think there should be a timeout, and then show the normal tray icon after that.

Reply:
Good point! In any case, my vote goes for shipping this patch and see how effective it is.